### PR TITLE
Add sha1.( support

### LINKED ISSUE #128: Support for the "(" class
Within certain formula, the "(" class is used for parenthetical notation.  I think that it's mainly (or perhaps only) when the parenthesis is the first item in the formula.  Regardless, the fix is pretty simple.  (I learn new things about R all the time; I had assumed that formula were made only of names and calls.)

``` r
library(digest)
sha1(a~(c+d))
#> Error: sha1() has no method for the '(' class
class((a~(c+d))[[3]])
#> [1] "("
"sha1.(" <- digest:::sha1.formula
sha1(a~(c+d))
#> [1] "3975451db3604aa3f8a947af38f29ed7d3930cb3"
```

<sup>Created on 2019-11-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by billdenney**
This fixes #128, but it has an issue with the documentation that appears to relate to the fact that the class name is "(".  I can't figure that one out with several tries for escaping it (nothing, quotes, and backslash all tried).

### CODE REVIEW

**Comment by eddelbuettel** [comment]
I am calling this a teachable moment. What is a dispatch on `(`? Do you have an example?  Is this for expressions "on the fly"?

File: man/sha1.rd [hunk 1]
```diff
@@ -90,5 +92,5 @@ with \code{sha1(x, algo = "sha1")}, they will be different starting from digest
 
 Until version 0.6.22, \code{sha1} ignored the attributes of the object for
 some classes. This was fixed in version 0.6.23. Use
-\code{option(sha1PackageVersion = 0.6.23)} to get the old behaviour.
+\code{option(sha1PackageVersion = 0.6.22)} to get the old behaviour.
```
Comment (eddelbuettel) [inline man/sha1.rd]: Good catch!

File: R/sha1.R [hunk 2]
```diff
@@ -449,3 +449,5 @@ sha1.formula <- function(x, digits = 14L, zapsmall = 7L, ..., algo = "sha1"){
     )
     digest(y, algo = algo)
 }
+
+"sha1.(" <- sha1.formula
```
Comment (eddelbuettel) [inline R/sha1.R]: So ... that line is the gist of the PR?  Kewl.

Comment (billdenney) [reply R/sha1.R]: Yep, it's pretty simple.

Comment (ThierryO) [reply R/sha1.R]: Elegant solution

**Pull request merged**